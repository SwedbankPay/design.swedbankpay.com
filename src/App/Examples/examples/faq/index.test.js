import React from "react";
import { shallow } from "enzyme";

import FAQ, { FAQSearch, FAQList, FAQSheet, FAQCard } from "./index";

describe("Examples: FAQ", () => {
    it("is defined", () => {
        expect(FAQ).toBeDefined();
    });

    it("renders", () => {
        const wrapper = shallow(<FAQ />);

        expect(wrapper).toMatchSnapshot();
    });

    describe("FAQSearch", () => {
        it("is defined", () => {
            expect(FAQSearch).toBeDefined();
        });

        it("renders", () => {
            const wrapper = shallow(<FAQSearch />);

            expect(wrapper).toMatchSnapshot();
        });
    });

    describe("FAQList", () => {
        it("is defined", () => {
            expect(FAQList).toBeDefined();
        });

        it("renders", () => {
            const wrapper = shallow(<FAQList />);

            expect(wrapper).toMatchSnapshot();
        });
    });

    describe("FAQSheet", () => {
        it("is defined", () => {
            expect(FAQSheet).toBeDefined();
        });

        it("renders", () => {
            const wrapper = shallow(<FAQSheet />);

            expect(wrapper).toMatchSnapshot();
        });
    });

    describe("FAQCard", () => {
        it("is defined", () => {
            expect(FAQCard).toBeDefined();
        });

        it("renders", () => {
            const wrapper = shallow(<FAQCard />);

            expect(wrapper).toMatchSnapshot();
        });
    });
});
