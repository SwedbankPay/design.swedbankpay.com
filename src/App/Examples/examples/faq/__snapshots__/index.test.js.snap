// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Examples: FAQ FAQCard renders 1`] = `
<div
  className="row justify-content-center mt-5"
>
  <div
    className="col-lg-8"
  >
    <div
      className="card card-primary"
    >
      <header
        className="card-header"
      >
        <h3>
          Further help
        </h3>
      </header>
      <div
        className="card-body d-flex flex-column"
      >
        <div
          className="media media-sm"
        >
          <div
            className="media-img"
          >
            <i
              aria-hidden="true"
              className="material-icons text-muted"
            >
              help
            </i>
          </div>
          <div
            className="media-body"
          >
            <h4
              className="text-muted"
            >
              Cannot find what you are looking for?
            </h4>
            <p>
              <small>
                Or any other inqueries?
              </small>
            </p>
          </div>
        </div>
        <p>
          Send in a question to get help from our support team
        </p>
        <button
          className="btn btn-primary btn-block"
          data-sheet-open="faq-sheet"
          type="button"
        >
          Send in a question
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Examples: FAQ FAQList renders 1`] = `
<div
  className="row justify-content-center"
>
  <div
    className="col-lg-8"
  >
    <Accordion
      items={
        Array [
          Object {
            "content": "Proident cupidatat irure excepteur id minim dolore mollit amet reprehenderit consectetur aliqua officia Lorem elit. Cupidatat ea tempor ut esse. Aliquip exercitation in nisi reprehenderit anim Lorem. Excepteur nostrud officia irure tempor aute ipsum. Voluptate nisi incididunt proident enim reprehenderit fugiat fugiat magna sit reprehenderit Lorem anim. Aute commodo cillum nisi consectetur amet eiusmod do tempor incididunt anim.",
            "id": "exp-1",
            "title": "Frequently asked question #1?",
          },
          Object {
            "content": "Proident cupidatat irure excepteur id minim dolore mollit amet reprehenderit consectetur aliqua officia Lorem elit. Cupidatat ea tempor ut esse. Aliquip exercitation in nisi reprehenderit anim Lorem. Excepteur nostrud officia irure tempor aute ipsum. Voluptate nisi incididunt proident enim reprehenderit fugiat fugiat magna sit reprehenderit Lorem anim. Aute commodo cillum nisi consectetur amet eiusmod do tempor incididunt anim.",
            "id": "exp-2",
            "title": "Frequently asked question #2?",
          },
          Object {
            "content": "Proident cupidatat irure excepteur id minim dolore mollit amet reprehenderit consectetur aliqua officia Lorem elit. Cupidatat ea tempor ut esse. Aliquip exercitation in nisi reprehenderit anim Lorem. Excepteur nostrud officia irure tempor aute ipsum. Voluptate nisi incididunt proident enim reprehenderit fugiat fugiat magna sit reprehenderit Lorem anim. Aute commodo cillum nisi consectetur amet eiusmod do tempor incididunt anim.",
            "id": "exp-3",
            "title": "Frequently asked question #3?",
          },
          Object {
            "content": "Proident cupidatat irure excepteur id minim dolore mollit amet reprehenderit consectetur aliqua officia Lorem elit. Cupidatat ea tempor ut esse. Aliquip exercitation in nisi reprehenderit anim Lorem. Excepteur nostrud officia irure tempor aute ipsum. Voluptate nisi incididunt proident enim reprehenderit fugiat fugiat magna sit reprehenderit Lorem anim. Aute commodo cillum nisi consectetur amet eiusmod do tempor incididunt anim.",
            "id": "exp-4",
            "title": "Frequently asked question #4?",
          },
          Object {
            "content": "Proident cupidatat irure excepteur id minim dolore mollit amet reprehenderit consectetur aliqua officia Lorem elit. Cupidatat ea tempor ut esse. Aliquip exercitation in nisi reprehenderit anim Lorem. Excepteur nostrud officia irure tempor aute ipsum. Voluptate nisi incididunt proident enim reprehenderit fugiat fugiat magna sit reprehenderit Lorem anim. Aute commodo cillum nisi consectetur amet eiusmod do tempor incididunt anim.",
            "id": "exp-5",
            "title": "Frequently asked question #5?",
          },
          Object {
            "content": "Proident cupidatat irure excepteur id minim dolore mollit amet reprehenderit consectetur aliqua officia Lorem elit. Cupidatat ea tempor ut esse. Aliquip exercitation in nisi reprehenderit anim Lorem. Excepteur nostrud officia irure tempor aute ipsum. Voluptate nisi incididunt proident enim reprehenderit fugiat fugiat magna sit reprehenderit Lorem anim. Aute commodo cillum nisi consectetur amet eiusmod do tempor incididunt anim.",
            "id": "exp-6",
            "title": "Frequently asked question #6?",
          },
        ]
      }
    />
  </div>
</div>
`;

exports[`Examples: FAQ FAQSearch renders 1`] = `
<Fragment>
  <div
    className="row justify-content-center"
  >
    <img
      alt="logo"
      className="examples-faq-logo"
      src="/TEST/designguide/assets/TESTBRAND-logo.svg"
    />
  </div>
  <div
    className="row justify-content-center"
  >
    <h2>
      How can we help?
    </h2>
  </div>
  <div
    className="row justify-content-center"
  >
    <div
      className="col-lg-8"
    >
      <InputGroup
        postfixBtnColor="primary"
        postfixType="button"
        postfixValue="Find"
        prefixType="icon"
        prefixValue="search"
        type="text"
      />
    </div>
  </div>
</Fragment>
`;

exports[`Examples: FAQ FAQSheet renders 1`] = `
<Sheet
  id="faq-sheet"
  requireAction={true}
>
  <h2>
    Ask a question
  </h2>
  <div
    className="form-group"
  >
    <div
      className="input-group"
    >
      <span
        className="input-group-addon"
      >
        <i
          aria-hidden="true"
          className="material-icons"
        >
          help
        </i>
      </span>
      <textarea
        className="form-control h-25"
        id="settings-description"
        type="textarea"
      />
    </div>
  </div>
  <button
    className="btn btn-primary"
    onClick={[Function]}
    type="button"
  >
    Send
  </button>
</Sheet>
`;

exports[`Examples: FAQ renders 1`] = `
<DocContainer>
  <ComponentPreview
    codeFigure={true}
    language="html"
    showCasePanel={true}
  >
    <div
      className="container"
    >
      <FAQSearch />
      <FAQList />
      <FAQSheet />
      <FAQCard />
    </div>
  </ComponentPreview>
</DocContainer>
`;
